sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/thetasketchesAggregates.scala

```diff
@@ -0,0 +1,682 @@
+   *   an Intersection instance wrapped with IntersectionAggregationBuffer
+   */
+  override def createAggregationBuffer(): ThetaSketchState = {
+    IntersectionAggregationBuffer(
+      SetOperation.builder
+        .setLogNominalEntries(lgNomEntries)
+        .buildIntersection)
+  }
+
+  /**
+   * Update the Intersection instance with the Compact sketch byte array obtained from the row.
+   *
+   * @param intersectionBuffer
+   *   A previously initialized Intersection instance
+   * @param input
+   *   An input row
+   */
+  override def update(
+      intersectionBuffer: ThetaSketchState,
+      input: InternalRow): ThetaSketchState = {
+    val v = left.eval(input)
+    v match {
+      case null => intersectionBuffer // Input is null, return buffer unchanged
+      case _ =>
+        left.dataType match {
+          case BinaryType =>
+            val memory = try {
+              Memory.wrap(v.asInstanceOf[Array[Byte]])
+            } catch {
+              case _: IllegalArgumentException | _: IndexOutOfBoundsException =>
+                throw QueryExecutionErrors.thetaInvalidInputSketchBuffer(prettyName)
+            }
+
+            val inputSketch = try {
+              CompactSketch.wrap(memory)
+            } catch {
+              case _: SketchesArgumentException =>
+                throw QueryExecutionErrors.thetaInvalidInputSketchBuffer(prettyName)
+            }
+
+            val intersection = intersectionBuffer match {
+              case IntersectionAggregationBuffer(existingIntersection) => existingIntersection
+              case _ => throw QueryExecutionErrors.thetaInvalidInputSketchBuffer(prettyName)
+            }
+            intersection.intersect(inputSketch)
+            IntersectionAggregationBuffer(intersection)
+          case _ =>
+            throw QueryExecutionErrors.thetaInvalidInputSketchBuffer(prettyName)
+        }
+    }
+  }
+
+  /**
+   * Merges an input Compact sketch into the Intersection which is acting as the aggregation
+   * buffer.
+   *
+   * @param intersectionBuffer
+   *   The Intersection instance used to store the aggregation result
+   * @param input
+   *   An input Intersection or Compact sketch instance
+   */
+  override def merge(
+      intersectionBuffer: ThetaSketchState,
+      input: ThetaSketchState): ThetaSketchState = {
+    (intersectionBuffer, input) match {
+      // Both are intersections, merge them directly
+      case (
+            IntersectionAggregationBuffer(intersectLeft),
+            IntersectionAggregationBuffer(intersectRight)) =>
+        intersectLeft.intersect(intersectRight.getResult)
+        IntersectionAggregationBuffer(intersectLeft)
+      // The input was serialized then deserialized
+      case (IntersectionAggregationBuffer(intersection), FinalizedSketch(sketch)) =>
+        intersection.intersect(sketch)
+        IntersectionAggregationBuffer(intersection)
+      // Should never make it here, but added cases for defensive programming
```

Comment:
super nit, could we please make every comment in the PR comprise complete sentences, each starting with a capital letter and ending with punctuation?